### DISCUSSION TITLE
I want an asset to depend on and load in a dbt model after its materialized

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17958] [Q&A] [answered]
Hi all - I’m trying to use a dbt-generated asset to create a Python one. I’m following <https://github.com/dagster-io/dagster/blob/8978f184b8b041e7f37fbdd886a1800d27fc87a2/examples/assets_dbt_python/assets_dbt_python/assets/forecasting/__init__.py#L20|this example>, and here is a snippet:
```@asset(
    ins={"my_dbt_model": AssetIn(key=get_asset_key_for_model([my_dbt_assets], model_name="my_dbt_model"))},
    compute_kind="python"
)
def my_python_asset(my_dbt_model: pd.DataFrame):
    df = my_dbt_model
    return df```
However, when I materialize the python asset in local Dagster (via dagster dev), that leads to the error: `FileNotFoundError: [Errno 2] No such file or directory: '/GitHub/my-repo/tmpnlfo8dgi/storage/my_dbt_model'` . What am I missing for referencing the dbt asset?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C04CW71AGBW/p1699890703735749?thread_ts=1699890703.735749&cid=C04CW71AGBW).*

**Comment by rexledesma** [comment, marked as answer]
You are missing an IO manager for your dbt assets: https://github.com/dagster-io/dagster/blob/8978f184b8b041e7f37fbdd886a1800d27fc87a2/examples/assets_dbt_python/assets_dbt_python/__init__.py#L36-L37

IO managers allow you to load assets as in-memory representations for downstream computations. [See the docs](https://docs.dagster.io/concepts/io-management/io-managers) for more details.

**Reply by skwon615** [reply]
Ah gotcha, thank you for that! Is the Duck DB Pandas IO manager the default one that you'd recommend for this use case?

**Reply by rexledesma** [reply]
In general, you should choose (or create) an IO manager that takes your storage backend and maps it to your desired Python object. 

If your dbt assets are stored in DuckDB and you want to manipulate your dbt models as pandas dataframes, then you should use the `DuckDBPandasIOManager`.

**Reply by skwon615** [reply]
Gotcha, that makes sense - thanks! And one more question - in the GitHub example project, where does it specify that the dbt assets use the DuckDB `io_manager`?

**Reply by rexledesma** [reply]
By default, if the `io_manager` is not specified, assets will use the `io_manager` defined by the resource key `"io_manager"`. 

So in the snippet that I provided above, we define the default IO manager for all assets. So the dbt assets definition uses the `DuckDBPandasIOManager`.

**Reply by skwon615** [reply]
Ah cool, so that resource key is the default. Awesome, will give this a shot - thank you!
